lib/factorial.rb

```diff
@@ -1,4 +1,17 @@
 # Computes factorial of the input number and returns it
 def factorial(number)
-  raise NotImplementedError
+  raise ArgumentError.new if number == nil
+  if number == 0 || number == 1
+    return 1
+  end
+
+  result = number
+  (number - 1).times do |i|
+    result *= number - (i + 1)
+  end
+  return result
 end
+
+
+# the time complexity for this solution is O(n), because the algorithm must repeat the multiplication loop (n - 1) times
```

Comment:
Minor: always explain what _n_ stands for. In this case, _n_ is the value of the input parameter, _number_